### ISSUE TITLE
Enforce good naming behavior in classes

### ISSUE THREAD

**Issue by Kavignon** [code-cracker/code-cracker#310]
Make sure that fields, properties and events are properly named.
First, make sure that they're not defined with a single character but also, that they respect the camelCase such as fields are to respect lowerCamelCase and properties are to respect UpperCamelCase.
Warning level : Warning
I don't know what diagnostics to give at the moment, so I'll say CCXXXX.

As for test cases, this is straightfoward. The analyzer will verify FieldDeclarationSyntax, EventDeclarationSyntax, PropertyDeclarationSyntax and evaluate their ISymbol to find out their name. Then, a simple Regex for each of those syntax will either validate or invalidate the name that was retrieved within the ISymbol. 

Basically, there would be six test case: one good and one bad foreach of the SyntaxNode that the analyzer has to evaluate.

**Comment by Kavignon** [comment]
No one has answered this thread. 
Is this a really bad idea ?

**Comment by giggio** [comment]
@Kavignon I just realized that this issue went unnoticed. I don't know why. Anyway, it is not a bad idea. But it is doing too much. How about we focus on fields and then create another one for fields? BTW, what styling guide would we follow?
Also, `CC0016` is taken. See: https://github.com/code-cracker/code-cracker/wiki/DiagnosticIds
You may update the wiki and bring the id here.

**Comment by giggio** [comment]
BTW, category is style, right?
Can you add test cases?

**Comment by sharwell** [comment]
@Kavignon We have a sample implementation and test code you may want to reference when you are implementing this in the StyleCopAnalyzers project. You can look for the [naming rules of interest](http://stylecop.com/docs/Naming%20Rules.html) by searching the [project code](https://github.com/DotNetAnalyzers/StyleCopAnalyzers) for the StyleCop diagnostic ID.

**Comment by Kavignon** [comment]
I shall write some tests and directly show them during the pull request ? 
Category: Style
@giggio  you said a first for fields and a second for fields :p
What could be done is a first for Fields and Properties, a second one for Classes and Struct, another for Interfaces and Enums?
I could do all of them !
@sharwell  I'm going to go through that list,but I thought we could reproduce what Resharper already does and look at it's naming rules !

**Comment by giggio** [comment]
I agree with category `Style`. But `Info` is better for severity.
Ok, we could do this issue as fields and properties then.
We still need test cases.

**Comment by giggio** [comment]
BTW, `CC0016` is taken. Check the diagnostic id wiki page: https://github.com/code-cracker/code-cracker/wiki/DiagnosticIds
You may take the next number and update the wiki.

**Comment by thiagolunardi** [comment]
And about sugest lowdash/underscore ('_') prefix for private fields?

**Comment by wmjordan** [comment]
@thiagolunardi I agree with you.
